src/overlays/actors/ovl_Bg_Ice_Turara/z_bg_ice_turara.c

```diff
@@ -27,23 +41,164 @@ const ActorInit Bg_Ice_Turara_InitVars = {
     (ActorFunc)BgIceTurara_Update,
     (ActorFunc)BgIceTurara_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Ice_Turara/BgIceTurara_Init.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Ice_Turara/BgIceTurara_Destroy.s")
+InitChainEntry sInitChain[] = {
+    ICHAIN_F32(uncullZoneScale, 600, ICHAIN_CONTINUE),
+    ICHAIN_F32(gravity, -3, ICHAIN_CONTINUE),
+    ICHAIN_F32(minVelocityY, -30, ICHAIN_CONTINUE),
+    ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_STOP),
+};
+
+// gravitational acceleration
+Vec3f D_8089267C[] = { 0.0f, -1.0f, 0.0f };
+
+Color_RGBA8 primColor[] = { 170, 255, 255, 255 };
+
+Color_RGBA8 envColor[] = { 0, 50, 100, 255 };
+
+void BgIceTurara_Init(Actor* thisx, GlobalContext* globalCtx) {
+    BgIceTurara* this = THIS;
+    s16 pad;
+    ColHeader* colHeader = NULL;
+
+    Actor_ProcessInitChain(&this->dyna.actor, sInitChain);
+    DynaPolyInfo_SetActorMove(&this->dyna, 0);
+    DynaPolyInfo_Alloc(&D_06002594, &colHeader);
+    Collider_InitCylinder(globalCtx, &this->collider);
+    Collider_SetCylinder(globalCtx, &this->collider, thisx, &sCylinderInit);
+    Collider_CylinderUpdate(thisx, &this->collider);
+    this->dyna.dynaPolyId = DynaPolyInfo_RegisterActor(globalCtx, &globalCtx->colCtx.dyna, thisx, colHeader);
```

Comment:
```suggestion
    Collider_CylinderUpdate(&this->actor, &this->collider);
    this->dyna.dynaPolyId = DynaPolyInfo_RegisterActor(globalCtx, &globalCtx->colCtx.dyna, &this->actor, colHeader);
```
Try not to use `thisx` unless needed for match.